README.md

```diff
@@ -1,169 +1,79 @@
 ```
 
-### Build Kubernetes
-
-kubeadm-nspawn needs the built Kubernetes binaries and hyperkube
-Docker image. You need to build them like that:
+`kubeadm-nspawn` will configure the networks it needs in `/etc/cni/net.d`.
 
 ```
-cd $GOPATH/src/k8s.io/kubernetes
-build/run.sh make
-cd cluster/images/hyperkube
-make VERSION=latest
-```
+# Build the tool
+$ make vendor all
 
-### Get CNI plugins
+# Get the recommended container image
+$ sudo machinectl pull-raw --verify=no https://stable.release.core-os.net/amd64-usr/current/coreos_developer_container.bin.bz2 coreos
 
-```
-go get -u github.com/containernetworking/plugins/plugins/main/bridge
-go get -u github.com/containernetworking/plugins/plugins/ipam/host-local
-```
+# Spawn and provision nodes for the cluster
+$ sudo GOPATH=$GOPATH CNI_PATH=$GOPATH/bin ./kubeadm-nspawn --kubernetes-version=1.7.0 up --image=coreos --nodes=3
 
-`kubeadm-nspawn` will configure the networks it needs in `/etc/cni/net.d`.
+# Setup Kubernetes
+$ sudo GOPATH=$GOPATH CNI_PATH=$GOPATH/bin ./kubeadm-nspawn --kubernetes-version=1.7.0 init
+```
 
-## Requirements
+## Run from local Kubernetes repository
 
-### on the host
+The requirements stay the same as above. Additionally you need Docker to run a local registry, where we will host
+the Hyperkube image of your Kubernetes build.
 
-  * systemd-nspawn v233, or systemd-nspawn v231 with backports for:
-    * `SYSTEMD_NSPAWN_USE_CGNS` https://github.com/systemd/systemd/pull/3809
-    * `SYSTEMD_NSPAWN_MOUNT_RW` and `SYSTEMD_NSPAWN_USE_NETNS` https://github.com/systemd/systemd/pull/4395
-  * glide from https://github.com/Masterminds/glide
+The following steps assume you have a local checkout of the Kubernetes source.
 
-## Build and run kubeadm-nspawn
+```
+# Build Kubernetes
+$ cd $GOPATH/src/k8s.io/kubernetes
+$ build/run.sh make
 
-Make sure you have `glide` available in you PATH.
-In the directory where you cloned this repository, please do:
+# Build a Hyperkube image
+$ cd cluster/images/hyperkube
+$ make VERSION=latest
+```
 
+Assuming you have built `kubeadm-nspawn` and pulled the CoreOS image, do:
 ```
-make vendor all
-sudo machinectl pull-raw --verify=no https://stable.release.core-os.net/amd64-usr/current/coreos_developer_container.bin.bz2 coreos
-sudo GOPATH=$GOPATH SYSTEMD_NSPAWN_PATH=<path_to_your_nspawn_binary> CNI_PATH=<path_to_cni_plugins_bins> ./kubeadm-nspawn up --image coreos --nodes 2
-sudo ./kubeadm-nspawn init
+# Spawn and provision nodes for the cluster
+$ sudo GOPATH=$GOPATH CNI_PATH=$GOPATH/bin ./kubeadm-nspawn up --image=coreos --nodes=3
+
+# Setup Kubernetes
+$ sudo GOPATH=$GOPATH CNI_PATH=$GOPATH/bin ./kubeadm-nspawn init
 ```
 
-Sometimes when Docker doesn't use the newest existing API, you may see
-the following error:
+## Test with Vagrant
 
-```
-2017/01/26 16:41:38 Error when pushing image: Error response from daemon: client is newer than server (client API version: 1.26, server API version: 1.24)
-```
+see [doc/vagrant](doc/vagrant.md)
```

Comment:
If there's a simpler way to test/use kube-spawn with less overhead in the future, that's great. But right now, the closest to a minimal, reproducible and working demo is the Vagrant setup as far as I can tell. In my opinion, proving that it works (e.g. https://github.com/kinvolk/kube-spawn/pull/52#discussion_r126647125) and making it testable (https://github.com/kinvolk/kube-spawn/issues/56) should be the first priority.